### PR TITLE
Added Keyboard Noodlepad_Micro

### PR DESCRIPTION
**Pull request by jessel92**
## Description

Added new KB to the `themadnoodle` folder `noodlepad_micro` running on the RP2040 chip
With via support

I also have other PRs Open for:
The rename and addition of another keyboard in `themadnoodle ` directory. 
PR Noodlepad Additions and Updates #22701

The addition of another Keyboard Udon13 in the same directory
Adding Keyboard The Mad Noodle Udon13 #22704

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/themadnoodle/noodlepad_micro/keymaps/via/rules.mk [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# This file is used to override the default rules.mk file
+WS2812_DRIVER = vendor
```
Comment (zvecr) [inline keyboards/themadnoodle/noodlepad_micro/keymaps/via/rules.mk:2]:
```suggestion
```

File: keyboards/themadnoodle/noodlepad_micro/keymaps/default/rules.mk [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# This file is used to override the default rules.mk file
+WS2812_DRIVER = vendor
+RGBLIGHT_DRIVER = ws2812
```
Comment (zvecr) [inline keyboards/themadnoodle/noodlepad_micro/keymaps/default/rules.mk:3]:
```suggestion
```

File: keyboards/themadnoodle/noodlepad_micro/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+     * |       |       |       |
+     * `-----------------------'
+     */
+    
+    [3] = LAYOUT_ortho_3x3(
+      KC_TRNS, KC_TRNS, KC_TRNS, 
+      KC_TRNS, KC_TRNS, KC_TRNS, 
+      KC_TRNS, KC_TRNS, KC_TRNS
+      )
+
```
Comment (zvecr) [inline keyboards/themadnoodle/noodlepad_micro/keymaps/default/keymap.c:73]: Dead layer
```suggestion
    )
```

File: keyboards/themadnoodle/noodlepad_micro/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+/*Encoder Mapping*/
+//-----------------------(ENC1)---------------------------------(ENC2)-----------------
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [0] =  { ENCODER_CCW_CW(KC_LEFT, KC_RGHT),      ENCODER_CCW_CW(KC_VOLD, KC_VOLU)  },
+    [1] =  { ENCODER_CCW_CW(RGB_HUD, RGB_HUI),      ENCODER_CCW_CW(RGB_SAD, RGB_SAI)  },
+    [2] =  { ENCODER_CCW_CW(RGB_VAD, RGB_VAI),      ENCODER_CCW_CW(RGB_SPD, RGB_SPI)  },
+    [3] =  { ENCODER_CCW_CW(KC_LEFT, KC_RGHT),      ENCODER_CCW_CW(KC_DOWN, KC_UP)    },
+
```
Comment (zvecr) [inline keyboards/themadnoodle/noodlepad_micro/keymaps/default/keymap.c:85]:
```suggestion
    [2] =  { ENCODER_CCW_CW(RGB_VAD, RGB_VAI),      ENCODER_CCW_CW(RGB_SPD, RGB_SPI)  }
```

File: keyboards/themadnoodle/noodlepad_micro/info.json [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+                {"matrix": [2, 0], "x": 0, "y": 2}
+            ]
+        }
+    },
+
+    "encoder": {
+        "enabled": true,
+        "rotary": [
+            { "pin_a": "GP2", "pin_b": "GP1", "resolution": 4 }
+            { "pin_a": "GP3", "pin_b": "GP4", "resolution": 4 }
```
Comment (zvecr) [inline keyboards/themadnoodle/noodlepad_micro/info.json:64]: Should not duplicate default settings
```suggestion
            { "pin_a": "GP2", "pin_b": "GP1" }
            { "pin_a": "GP3", "pin_b": "GP4" }
```

File: keyboards/themadnoodle/noodlepad_micro/config.h [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+//#define NO_ACTION_ONESHOT
+
+
+/* RGB BackLight */
+#define WS2812_PIO_USE_PIO1
+#define WS2812_DI_PIN GP29
+ 
+/*RGB Defaults*/
+#define RGBLIGHT_DEFAULT_MODE RGBLIGHT_MODE_RAINBOW_SWIRL + 5
+/* See info.json for more RGB settings*/
```
Comment (zvecr) [inline keyboards/themadnoodle/noodlepad_micro/config.h:29]:
```suggestion
#define RGBLIGHT_DEFAULT_MODE RGBLIGHT_MODE_RAINBOW_SWIRL + 5
```

File: keyboards/themadnoodle/noodlepad_micro/info.json [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+                {"matrix": [1, 2], "x": 2, "y": 1},
+                {"matrix": [1, 1], "x": 1, "y": 1},
+                {"matrix": [1, 0], "x": 0, "y": 1},
+                {"matrix": [2, 2], "x": 2, "y": 2},
+                {"matrix": [2, 1], "x": 1, "y": 2},
+                {"matrix": [2, 0], "x": 0, "y": 2}
+            ]
+        }
+    },
+
```
Comment (zvecr) [inline keyboards/themadnoodle/noodlepad_micro/info.json:59]:
```suggestion
    "ws2812": {
        "pin": "GP29",
        "driver": "vendor"
    },
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/themadnoodle/noodlepad_micro/info.json [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    "usb": {
+        "device_version": "3.0.0",
+        "pid": "0x0004",
+        "vid": "0x6A6C"
+    },
+    "layouts": {
+        "LAYOUT_ortho_3x3": {
+            "layout": [
+                {"matrix": [0, 2], "x": 2, "y": 0},
+                {"matrix": [0, 1], "x": 1, "y": 0},
```
Comment (zvecr) [inline keyboards/themadnoodle/noodlepad_micro/info.json:48]:
```suggestion
        "LAYOUT": {
            "layout": [
                {"matrix": [0, 2], "x": 2, "y": 0},
```

File: keyboards/themadnoodle/noodlepad_micro/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+     * |       |       |       |  
+     * |-------+-------+-------|
+     * |       |       |       |
+     * |-------+-------+-------|
+     * |       |       |       |
+     * `-----------------------'
+     */
+    
+    [2] = LAYOUT_ortho_3x3(
+      KC_TRNS, KC_TRNS, KC_TRNS,
```
Comment (zvecr) [inline keyboards/themadnoodle/noodlepad_micro/keymaps/default/keymap.c:53]:
```suggestion
    [0] = LAYOUT(
      KC_MPRV,           KC_MNXT, 
      KC_MSTP, KC_MPLY, KC_MSEL,
      LT(2,KC_CALC), KC_MAIL, LT(1, KC_MYCM)
      ),


    /* LAYER 1
     * ,--ENC2--       --ENC1--.
     * | MODE+ |       | MODE- |  
     * |-------+-------+-------|
     * |Bright-|  Tog  |Bright+|
     * |-------+-------+-------|
     * | PLAIN |BREATH |       |
     * `-----------------------'
     */
    
    [1] = LAYOUT(
      RGB_MOD,          RGB_RMOD, 
      RGB_VAD, RGB_TOG, RGB_VAI, 
      RGB_M_P, RGB_M_B, KC_TRNS
      ),

      
    /* LAYER 2 (ENCODER)
     * ,--ENC2--       --ENC1--.
     * |       |       |       |  
     * |-------+-------+-------|
     * |       |       |       |
     * |-------+-------+-------|
     * |       |       |       |
     * `-----------------------'
     */
    
    [2] = LAYOUT(
      KC_TRNS,          KC_TRNS, 
```

File: keyboards/themadnoodle/noodlepad_micro/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+     * |       |       |       |  
+     * |-------+-------+-------|
+     * | TO(1) |       | TO(2) |
+     * |-------+-------+-------|
+     * |       |       | TO(0) |
+     * `-----------------------'
+     */
+    
+    [3] = LAYOUT_ortho_3x3(
+      KC_TRNS, KC_TRNS, KC_TRNS,
```
Comment (zvecr) [inline keyboards/themadnoodle/noodlepad_micro/keymaps/via/keymap.c:69]:
```suggestion
    [0] = LAYOUT(
      KC_MPRV,           KC_MNXT, 
      KC_MSTP, KC_MPLY, KC_MSEL,
      KC_CALC, KC_MYCM, TO(3)
      ),


    /* LAYER 1
     * ,--ENC2--       --ENC1--.
     * | MODE+ |       | MODE- |  
     * |-------+-------+-------|
     * |Bright-|  Tog  |Bright+|
     * |-------+-------+-------|
     * | PLAIN |BREATH | TO(0) |
     * `-----------------------'
     */
    
    [1] = LAYOUT(
      RGB_MOD,          RGB_RMOD, 
      RGB_VAD, RGB_TOG, RGB_VAI, 
      RGB_M_P, RGB_M_B, TO(0)
      ),

      
    /* LAYER 2 
     * ,--ENC2--       --ENC1--.
     * |       |       |       |  
     * |-------+-------+-------|
     * |       |       |       |
     * |-------+-------+-------|
     * |       |       | TO(0) |
     * `-----------------------'
     */
    
    [2] = LAYOUT(
      KC_TRNS,          KC_TRNS, 
      KC_TRNS, KC_TRNS, KC_TRNS, 
      KC_TRNS, KC_TRNS, TO(0)
      ),

    /* LAYER 3 
     * ,--ENC2--       --ENC1--.
     * |       |       |       |  
     * |-------+-------+-------|
     * | TO(1) |       | TO(2) |
     * |-------+-------+-------|
     * |       |       | TO(0) |
     * `-----------------------'
     */
    
    [3] = LAYOUT(
      KC_TRNS,         KC_TRNS, 
```

**Review by zvecr — CHANGES_REQUESTED** [review]
Product images shows that it does not have 9 keys

File: keyboards/themadnoodle/noodlepad_micro/info.json [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+                {"matrix": [2, 0], "x": 0, "y": 2}
+            ]
+        }
+    },
+    "ws2812": {
+        "pin": "GP29",
+        "driver": "vendor"
+    },
+    "encoder": {
+        "enabled": true,
```
Comment (waffle87) [inline keyboards/themadnoodle/noodlepad_micro/info.json:63]:
```suggestion
```

File: keyboards/themadnoodle/noodlepad_micro/info.json [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+    "bootloader": "rp2040",
+    "diode_direction": "ROW2COL",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgblight": true
```
Comment (waffle87) [inline keyboards/themadnoodle/noodlepad_micro/info.json:15]:
```suggestion
        "rgblight": true,
        "encoder": true
```

File: keyboards/themadnoodle/noodlepad_micro/keymaps/via/rules.mk [hunk 13]
```diff
@@ -0,0 +1,3 @@
+RGBLIGHT_DRIVER = ws2812
```
Comment (waffle87) [inline keyboards/themadnoodle/noodlepad_micro/keymaps/via/rules.mk:1]:
```suggestion
```

File: keyboards/themadnoodle/noodlepad_micro/readme.md [hunk 14]
```diff
@@ -0,0 +1,30 @@
+# NoodlePad [Micro]
+
+![NoodPad[Mircro]](https://static.wixstatic.com/media/59d0ff_a93e3066685e4273b4c628e702ba981d~mv2.jpg)
```
Comment (waffle87) [inline keyboards/themadnoodle/noodlepad_micro/readme.md:3]: Please rehost this image on Imgur.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/themadnoodle/noodlepad_micro/info.json [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+        "LAYOUT": {
+            "layout": [
+                {"matrix": [0, 2], "x": 2, "y": 0},
+                {"matrix": [0, 0], "x": 0, "y": 0},
+                {"matrix": [1, 2], "x": 2, "y": 1},
+                {"matrix": [1, 1], "x": 1, "y": 1},
+                {"matrix": [1, 0], "x": 0, "y": 1},
+                {"matrix": [2, 2], "x": 2, "y": 2},
+                {"matrix": [2, 1], "x": 1, "y": 2},
+                {"matrix": [2, 0], "x": 0, "y": 2}
```
Comment (zvecr) [inline keyboards/themadnoodle/noodlepad_micro/info.json:55]: The order here is quite odd when compared to the layout macros. Right to left. The order should be left to right, updating matrix_pin order might also help keep the matrix values logical.

File: keyboards/themadnoodle/noodlepad_micro/readme.md (hunk 14 above)
Comment (jessel92) [reply keyboards/themadnoodle/noodlepad_micro/readme.md]: Done!

**Review by jessel92 — COMMENTED** [review]
All Should be good now i think

File: keyboards/themadnoodle/noodlepad_micro/readme.md [hunk 16]
```diff
@@ -0,0 +1,30 @@
+# NoodlePad [Micro]
+
+![NoodPad[Mircro]](https://imgur.com/a/l1iZjWX)
```
Comment (sigprof) [inline keyboards/themadnoodle/noodlepad_micro/readme.md:3]:
```suggestion
![NoodlePad [Micro]](https://i.imgur.com/uRmVt3ah.jpg)
```

File: keyboards/themadnoodle/noodlepad_micro/readme.md [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+
+* Keyboard Maintainer: [The Mad Noodle](https://github.com/The-Mad-Noodle)
+* Hardware Supported: NoodlePad [Micro]
+* Hardware Availability: https://www.madnoodleprototypes.com/shop
+
+This folder is set as default in the parent rules.mk - it will build firmware compatible with the RP2040 Community Edition. See [here](https://docs.qmk.fm/#/platformdev_rp2040?id=rp2040_ce) for a list.
+
+Compile example for this keyboard (after setting up your build environment):
+
+    qmk compile -kb themadnoodle/noodlepad/micro -km default
```
Comment (sigprof) [inline keyboards/themadnoodle/noodlepad_micro/readme.md:15]:
```suggestion
    qmk compile -kb themadnoodle/noodlepad_micro -km default
```

File: keyboards/themadnoodle/noodlepad_micro/readme.md [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+
+Flashing example for this keyboard:
+
+    qmk flash -kb themadnoodle/noodlepad/micro -km default
```
Comment (sigprof) [inline keyboards/themadnoodle/noodlepad_micro/readme.md:19]:
```suggestion
    qmk flash -kb themadnoodle/noodlepad_micro -km default
```

File: keyboards/themadnoodle/noodlepad_micro/readme.md [hunk 19]
```diff
@@ -0,0 +1,30 @@
+# NoodlePad [Micro]
+
+![NoodPad[Mircro]](https://imgur.com/a/l1iZjWX)
+
+The Noodlepad [Micro] is a 6 key 2 encoder macro keypad designed using RP2040 chipset
```
Comment (sigprof) [inline keyboards/themadnoodle/noodlepad_micro/readme.md:5]:
```suggestion
The NoodlePad [Micro] is a 6 key 2 encoder macro keypad designed using RP2040 chipset.
```

File: keyboards/themadnoodle/noodlepad_micro/readme.md [hunk 20]
```diff
@@ -0,0 +1,30 @@
...
+
+* Keyboard Maintainer: [The Mad Noodle](https://github.com/The-Mad-Noodle)
+* Hardware Supported: NoodlePad [Micro]
+* Hardware Availability: https://www.madnoodleprototypes.com/shop
+
+This folder is set as default in the parent rules.mk - it will build firmware compatible with the RP2040 Community Edition. See [here](https://docs.qmk.fm/#/platformdev_rp2040?id=rp2040_ce) for a list.
+
```
Comment (sigprof) [inline keyboards/themadnoodle/noodlepad_micro/readme.md:12]:
```suggestion
```
This looks like remnants from some older text which is not relevant for this board.

**Review by sigprof — CHANGES_REQUESTED** [review]

File: keyboards/themadnoodle/noodlepad_micro/readme.md (hunk 19 above)
Comment (jessel92) [reply keyboards/themadnoodle/noodlepad_micro/readme.md:5]: Hahaha I really need to work on my spell checking 😅

**Review by waffle87 — APPROVED** [review]

**Comment by jessel92** [comment]
@sigprof  @zvecr I'm Just checking in to see if you got a chance to review this? I'm on a time crunch with this new keypad in particular.

**Comment by waffle87** [comment]
copy/pasta from discord:
> While we know that your PR is important to you, it may take some time for QMK collaborators to get to it.
> PR reviews are something that is done in their free time -- collaborators are not paid nor compensated for the time spent reviewing, as it is a labor of love. As such, this means that it can take time for them to get to your PR -- things like family, or life can get in the way of finding time to perform reviews, and burnout is a serious consideration. The QMK firmware repository averages 300 PRs opened and 300 PRs merged every month, so please have patience.
> Also, anyone can review PRs -- adding meaningful reviews to other PRs is a great way to contribute back. This helps reduce the burden on collaborators, meaning they may be able to get to your PR that much faster.

Mind you, this PR moved extremely quickly on the scale of things.

**Comment by jessel92** [comment]
@waffle87 totally understood! I guess noobness is just showing hahaha. 
I will wait patiently!

**Review by drashna — APPROVED** [review]

**Pull request merged**